src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DocumentPresentation/TextDocumentUriPresentationEndpoint.cs

```diff
@@ -131,25 +131,16 @@ protected override IRazorPresentationParams CreateRazorRequestParameters(UriPres
     {
         _logger.LogInformation("Trying to find document info for dropped uri {uri}.", uri);
 
-        var documentSnapshot = await _projectSnapshotManagerDispatcher.RunOnDispatcherThreadAsync(() =>
-        {
-            if (_documentResolver.TryResolveDocument(uri.GetAbsoluteOrUNCPath(), out var documentSnapshot))
-            {
-                return documentSnapshot;
-            }
-
-            return null;
-        }, cancellationToken).ConfigureAwait(false);
-
-        if (documentSnapshot is null)
+        var documentContext = await _documentContextFactory.TryCreateAsync(uri, cancellationToken).ConfigureAwait(false);
```

Comment:
This is one of the spots where there was an issue. This code used to use `TryCreateAsync`, but that meant you could only drag components that were open in VS. That was fixed, by changing to use `DocumentResolver`, but now that `TryCreateAsync` does what you would expect, the previous code can be put back. I think you'll agree, its a nicer API.